### ISSUE TITLE
Improvement in message for non-existing package on different platform

### ISSUE THREAD

**Issue by nehaljwani** [conda/conda#6054]
If on platform X, I search for a package on platform Y, and the package doesn't exist for platform Y, then the output message says it searched for packages in channels for platform X, instead of platform Y
```
(root) [nwani@centos6x64 aggregate]$ conda search non-existing-package --platform win-64
Fetching package metadata ...........

PackageNotFoundError: Packages missing in current channels:
            
  - non-existing-package

We have searched for the packages in the following channels:
            
  - https://repo.continuum.io/pkgs/main/linux-64
  - https://repo.continuum.io/pkgs/main/noarch
  - https://repo.continuum.io/pkgs/free/linux-64
  - https://repo.continuum.io/pkgs/free/noarch
  - https://repo.continuum.io/pkgs/r/linux-64
  - https://repo.continuum.io/pkgs/r/noarch
  - https://repo.continuum.io/pkgs/pro/linux-64
  - https://repo.continuum.io/pkgs/pro/noarch
```
I'd expect the output to be:
```
Fetching package metadata .............

PackageNotFoundError: Packages missing in current channels:

  - non-existing-package

We have searched for the packages in the following channels:

  - https://repo.continuum.io/pkgs/main/win-64
  - https://repo.continuum.io/pkgs/main/noarch
  - https://repo.continuum.io/pkgs/free/win-64
  - https://repo.continuum.io/pkgs/free/noarch
  - https://repo.continuum.io/pkgs/r/win-64
  - https://repo.continuum.io/pkgs/r/noarch
  - https://repo.continuum.io/pkgs/pro/win-64
  - https://repo.continuum.io/pkgs/pro/noarch
  - https://repo.continuum.io/pkgs/msys2/win-64
  - https://repo.continuum.io/pkgs/msys2/noarch
```

**Comment by kalefranz** [comment]
`conda search` changes slightly between conda 4.3 and conda 4.4.  A fix of this nature should specifically be targeted at 4.4.  Is that what you were running on here?

**Comment by kalefranz** [comment]
Fixed as of 4.4.0rc2

**Comment by kalefranz** [comment]
```
$ git checkout 4.4.0rc2

$ python -m conda search non-existing-package --platform win-64
Loading channels: done

PackagesNotFoundError: The following packages are not available from current channels:

  - non-existing-package

Current channels:

  - https://repo.continuum.io/pkgs/main/win-64
  - https://repo.continuum.io/pkgs/main/noarch
  - https://repo.continuum.io/pkgs/free/win-64
  - https://repo.continuum.io/pkgs/free/noarch
  - https://repo.continuum.io/pkgs/r/win-64
  - https://repo.continuum.io/pkgs/r/noarch
  - https://repo.continuum.io/pkgs/pro/win-64
  - https://repo.continuum.io/pkgs/pro/noarch

```

ran on macOS
